src/lib/components/card/Card.component.js

```diff
@@ -0,0 +1,105 @@
+// @flow
+import React from "react";
+import styled, { css } from "styled-components";
+import type { Node } from "react";
+import { getTheme, getThemePropSelector, hex2RGB } from '../../utils';
+import * as defaultTheme from '../../style/theme';
+
+type Props = {
+  width?: string,
+  height?: string,
+  header?: Node,
+  children?: Node,
+  headerBackgroundColor?: string,
+  childrenBackgroundColor?: string,
+  borderColor?: string,
+  onClick?: any => void
+};
+
+const CardContainer = styled.div`
+  display: flex;
+  flex-flow: column;
+  color: ${getThemePropSelector("textPrimary")};
+  height: ${props => props.height};
+  width: ${props => props.width};
+  margin: ${defaultTheme.padding.small};
+  border: 2px solid transparent;
+  border-collapse: collapse;
+  box-sizing: border-box;
+
+  &:hover {
+    ${props => {
+      const borderColor = getTheme(props)[props.borderColor]
+      return css`
+        border: 2px solid ${borderColor};
+        border-radius: 8px;
+      `;
+    }}
+  }
+  
+  &:hover > div {
+    background-image: linear-gradient(rgba(${props => hex2RGB(getTheme(props)[props.borderColor]).join(',')}, 0.05) 0 0);
+  }
```

Comment:
I think it is missing an active and focus state as well as keyboard navigation. You can get inspired from this gist https://gist.github.com/JBWatenbergScality/b5975905defdcab9c4300da2f7e19cc5